### PR TITLE
(DOCSP-21714) Highlight v10 connector on landing page

### PR DESCRIPTION
**Pull request by zach-carr**
## Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-spark-connector/blob/master/REVIEWING.md)

I'll update the dochub link when the blog post goes live. This is the draft: https://docs.google.com/document/d/1qLne4CYIezgFfeNITaxRH-MfVIlIqUjZBHW3xCBc07s/edit

Dochub link not working for me at the moment, investigating that. 

`{"_id":{"$oid":"6245ba2ae4def541c65c2c66"},"url":"https://www.mongodb.com/docs/spark-connector/current/","name":"spark-connector-v10-announcement-blog"}`

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-21714

### Snooty build log:
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=6245bcc92e8f69777c58b39f

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/spark-connector/docsworker-xlarge/DOCSP-21714/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [ ] Are all the links working?
- [x] Are the staging and workerpool job links in the PR description updated?

### If your page documents a concept, does it meet the following criteria?

- [ ] Target the [Jasmin persona](https://drive.google.com/file/d/14FbBOLCVxwSP6M9BK4Nz1Ir9tzxT8_02/view)
- [ ] Target the [Lucas persona](https://drive.google.com/file/d/1J2vqJxo7ldv7OP_obA9Q-avf0o_ju4Lk/view)

### CODE REVIEW

File: source/index.txt [hunk 1]
```diff
@@ -8,6 +8,22 @@ The `MongoDB Connector for Spark
 <https://www.mongodb.com/products/spark-connector>`_ provides
 integration between MongoDB and Apache Spark.
 
+.. note::
+
+   Version 10.x of the MongoDB Connector for Spark is an all-new 
+   connector based on the latest Spark API. Install and migrate to 
+   version 10.x to take advantage of new capabilities, such as tighter 
+   integration with Spark 
+   :ref:`Structured Streaming <spark-structured-streaming>`.
```
Comment (JuliaMongo) [inline source/index.txt:17]: Can we move the word Spark into the ref itself? As follows:
:ref:`Spark Structured Streaming <spark-structured-streaming>`.

File: source/index.txt [hunk 2]
```diff
@@ -8,6 +8,22 @@ The `MongoDB Connector for Spark
...
+
+   Version 10.x uses a different namespace,
```
Comment (JuliaMongo) [inline source/index.txt:19]: Suggestion:
Version 10.x uses a different namespace, 
   ``com.mongodb.spark.sql.connector.MongoTableProvider``.
   This allows you to use old versions of the MongoDB connector 
   (versions 3.x and earlier) in parallel with version 10.x.

**Review by JuliaMongo — APPROVED** [review]
LGTM, with a few small comments. Thank you, Zach!

File: source/index.txt [hunk 3]
```diff
@@ -8,6 +8,22 @@ The `MongoDB Connector for Spark
 <https://www.mongodb.com/products/spark-connector>`_ provides
 integration between MongoDB and Apache Spark.
 
+.. note::
+
+   Version 10.x of the MongoDB Connector for Spark is an all-new 
+   connector based on the latest Spark API. Install and migrate to
```
[the remark this answers is not in the document]
Comment (zach-carr) [reply source/index.txt:14]: For context, this is a new connector that is completely separate from the old one. There will probably be some confusion about that so I want to make it as clear as possible. 

Planning to leave as-is, it's important for customers to understand this is a new connector and doesn't have an upgrade path - you need to install a whole new thing.

File: source/index.txt (hunk 1 above)
Comment (zach-carr) [reply source/index.txt:17]: Done

File: source/index.txt (hunk 2 above)
Comment (zach-carr) [reply source/index.txt:19]: That's better, done 👍

File: source/index.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (zach-carr) [reply source/index.txt:14]: I'm going to wait to make this change to see what Robert (the PM) might have to say about the wording. It's especially important to highlight that it's new (see context in another comment)

**Pull request merged**